### PR TITLE
feat(react-renderer): Support renderToReadableStream options for hydration

### PR DESCRIPTION
**Pull request by yoshikouki**
### Features

- Adds support for `renderToReadableStream` options in `(react-renderer)`
- Enables more flexible and performance-optimized hydration strategies

### Benefits

- Using react-renderer brings interaction to our hono application and also supports [\<Suspense\>](https://react.dev/reference/react/Suspense)
- Allows control of [render process options](https://react.dev/reference/react-dom/server/renderToReadableStream#parameters)

### Examples of use

https://github.com/yoshikouki/hono-spa-react-bun/blob/46208aa2ffa6c9375ccea14e9d724496850e336c/src/index.tsx#L29-L37

```ts
// src/index.tsx

app.get(
  "*",
  reactRenderer(Layout, {
    stream: true,
    readableStreamOptions: {
      bootstrapScripts: ["/src/client.tsx"],
    },
  })
);

app.get("/", (c) => {
  return c.render(<App />);
});
```

```ts
// src/client.tsx
import { hydrateRoot } from "react-dom/client";
import { App, Layout } from "./app";

hydrateRoot(
  document,
  <Layout>
    <App />
  </Layout>
);
```

### Ref:

https://react.dev/reference/react-dom/server/renderToReadableStream#rendering-a-react-tree-as-html-to-a-readable-web-stream

### CODE REVIEW

File: .changeset/good-hairs-compare.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@hono/react-renderer': patch
```
Comment (yusukebe) [inline .changeset/good-hairs-compare.md:2]: This should be `minor` instead of `patch`!

**Comment by yusukebe** [comment]
Hi @yoshikouki !

This is a good feature! I've left one comment. Thanks!

File: .changeset/good-hairs-compare.md (hunk 1 above)
Comment (yoshikouki) [reply .changeset/good-hairs-compare.md]: Sure!

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yoshikouki** [comment]
Thanks for your review!

**Pull request merged**